homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,210 @@
+"""Sensor configuration for VegeHub integration."""
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfElectricPotential, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CHAN_TYPE_BATTERY,
+    CHAN_TYPE_SENSOR,
+    DOMAIN,
+    MANUFACTURER,
+    MODEL,
+    OPTION_DATA_TYPE_CHOICES,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Vegetronix sensors from a config entry."""
+    sensors = []
+    mac_address = str(config_entry.data.get("mac_address"))
+    ip_addr = str(config_entry.data.get("ip_addr"))
+    num_sensors = int(config_entry.data.get("hub", {}).get("num_channels") or 0)
+    is_ac = int(config_entry.data.get("hub", {}).get("is_ac") or 0)
+
+    # We add up the number of sensors, plus the number of actuators, then add one
+    # for battery reading, and one because the array is 1 based instead of 0 based.
+    for i in range(num_sensors + 1):  # Add 1 for battery
+        if i > num_sensors:  # Now we're into actuators
+            continue  # Those will be taken care of by switch.py
+
+        if i == num_sensors and is_ac:
+            # Skipping battery slot for AC hub
+            continue
+
+        name = f"VegeHub Sensor {i + 1}"
+
+        chan_type = CHAN_TYPE_SENSOR
+        if i == num_sensors:
+            name = "Battery"
+            chan_type = CHAN_TYPE_BATTERY
+
+        sensor = VegeHubSensor(
+            name=name,
+            mac_address=mac_address,
+            slot=i + 1,
+            ip_addr=ip_addr,
+            dev_name=str(config_entry.data.get("hostname")),
+            data_type=str(config_entry.options.get(f"data_type_{i + 1}", None)),
+            chan_type=chan_type,
+        )
+
+        hass.data[DOMAIN][sensor.unique_id] = sensor
+
+        sensors.append(sensor)
+
+    if sensors:
+        async_add_entities(sensors)
+
+
+class VegeHubSensor(SensorEntity):
+    """Class for VegeHub Analog Sensors."""
+
+    def __init__(
+        self,
+        name: str,
+        mac_address: str,
+        slot: int,
+        ip_addr: str,
+        dev_name: str,
+        data_type: str,
+        chan_type: str,
+    ) -> None:
+        """Initialize the sensor."""
+        new_id = (
+            f"vegehub_{mac_address}_{slot}".lower()
+        )  # Generate a unique_id using mac and slot
+
+        self._attr_name: str = name
```

Comment:
set _attr_has_entity_name = True (required for new integrations) and a translation_key and set the actual name in strings.json to make the names translateable

Reply:
I added _attr_has_entity_name and translation_key in commit 166f2c9db70cd478e4aaafbf52c366b802ca4781, but I can't figure out how I would be able to reference a name that I had defined in strings.json.

I would need to be able to concatenate on a number on to it while creating sensors in sensor.py, because the name changes based on how many sensors there are on the target VegeHub. Is there a function that I am not seeing that allows me to get a translated string defined in strings.json?

Reply:
you can set `self._attr_translation_placeholders = {"key": "mykey"}` and in the strings add a placeholder like `"My Sensor {key}"`

Reply:
Excellent, thank you. I made that change. I appreciate your patience with my lack of knowledge on this. Commit 0d05cdba8fd21a0cee24416cfa05c59482168e4d has the change.